homeassistant/components/esphome/config_flow.py

```diff
@@ -86,6 +86,10 @@ async def async_step_reauth_confirm(
                 return await self._async_authenticate_or_add()
             errors["base"] = error
 
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert entry is not None
+        self._name = entry.title
```

Comment:
Isn't it easier to set this in `async_step_reauth`?
As it has the full entry available already...

Reply:
:man_facepalming: 

Still need to understand HA's code more =)